tests/connect.php

```diff
@@ -110,6 +119,10 @@ public function should_close_connection_after_configured_amount_of_failed_reconn
             $closed->resolve(true);
         });
 
+        // Need a watcher to keep the loop running after connection is closed.
+        $watcher = Loop::delay(180000, function (): void {
```

Comment:
This is highly suspicious. I never encountered a scenario where this would be needed.

Reply:
Not at all. Let me explain. In this test you're creating a connection which can't succeed and will fail with timeout, yes? This means that afterwards there is no active watcher anymore and the loop would just end immediately. But you still want to use the broken connection in order to call `appendToStreamAsync` and expect it to throw `InvalidOperationException`.

Reply:
It's only needed for this particular test case. In a real life application you would be okay with the loop ending immediately. But we still need the test because again in real life application the loop might stay active because of something else entirely.